packages/sync/src/class-users.php

```diff
@@ -44,13 +43,23 @@ class Users {
 	 * @todo Eventually, connection needs to be instantiated at the top level in the sync package.
 	 */
 	public static function init() {
-		self::$connection = new Jetpack_Connection();
+		add_action( 'jetpack_loaded', array( __CLASS__, 'on_jetpack_loaded' ) );
```

Comment:
Are we certain that the connection object will not be necessary in this class before `plugins_loaded`? It seems that a test is failing precisely because it tries to access the connection object earlier.

Reply:
Yes, that's the point of this research. We need to make sure that everything that runs before `plugins_loaded` has a legitimate reason to do so. I'll work on that too, thank you!

Reply:
It looks like `Users::on_jetpack_loaded()` isn’t being called when the ‘jetpack_loaded’ action fires because `Users::init()` is called after `jetpack_loaded` has already fired.

`Users::init()` is hooked onto the `plugins_loaded` action with a priority of 90 (see [`Actions::init()`](https://github.com/Automattic/jetpack/blob/master/packages/sync/src/class-actions.php#L67) which is called in `Jetpack::late_initialization()`). The `jetpack_loaded` action fires in the `after_plugins_loaded()` method, which is hooked onto the `plugins_loaded`  action with a priority of 10.

Reply:
I have fixed that in one of the later commits, thanks!